msg-socket/src/req/driver.rs

```diff
@@ -97,6 +119,9 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Future for ReqDriver<T> {
                 Poll::Pending => {}
             }
 
+            // Check for request timeouts
+            this.check_timeouts();
```

Comment:
We still have the same problem here of checking every request every time the driver is polled. You should use `tokio::time::Interval` to only check this on a pre-defined interval (i.e. check if `interval.poll_tick().is_ready()`). This interval will call the waker at the correct times.

The interval duration should be 1/10th of the configured timeout.